core/app/models/spree/product.rb

```diff
@@ -274,6 +274,10 @@ def find_variant_property_rule(option_value_ids)
       end
     end
 
+    def cache_key
+      super + "/prices/" + prices.currently_valid.pluck(:id).join("/")
```

Comment:
Nope, coming to think of it: This PR should really just introduce this feature. `Variant#prices.currently_valid` will return all prices (all currencies and so on), and custom pricers will just add more attributes to select prices from - this caching strategy would still work in most cases. I'll work on your suggestions (use `timecop` and use the `change` matcher).